File: q3/vocab.py
f1 = open('20ng-rec_talk.txt','r')

f2 = open('vocab.txt','w')

#the dictionary
d = {}
words = []
count = 0
num=0
for line in f1:
	line = line.strip()
	line = line.split()
	line = line[1:]
	count = count + 1
	if((count%500)==0):
	 print count
	for i in range(0,len(line)):
		if(not(d.has_key(line[i]))):
			d[line[i]] = num
			num = num + 1

print num

words = d.keys()
words.sort()
print "uniqie words......\n"
l = len(words)
print l

for i in range (0,l):
	s = words[i]+'\n'
	f2.write(s)

File: q3/vocab.m
v = containers.Map

f1 = fopen('vocab2.txt','w');

num=1;
f2 = fopen('20ng-rec_talk.txt');
line_no=0;
while ~feof(f2)
line_no=line_no+1
line = fgetl(f2);
a = strsplit(line,' ');
b=strsplit(a{1},'\t');
v(b{2})=num;
num=num+1;

for i=1:length(a)
	if(not(isKey(v,a{i})))
	v(a{i}) = num;
	num=num+1;
	end
end
end
z=keys(v);

for i=1:length(z)
	fprintf(f1,'%s\n',z{i});
end
